config/fcos/v1_5_exp/translate_test.go

```diff
@@ -1487,3 +1487,195 @@ func TestTranslateExtensions(t *testing.T) {
 		})
 	}
 }
+
+// TestTranslateGrub tests translating the Butane config Grub section.
+func TestTranslateGrub(t *testing.T) {
+	tests := []struct {
+		in         Config
+		out        types.Config
+		exceptions []translate.Translation
+		report     report.Report
+	}{
+		// config with 1 user
+		{
+			Config{
+				Grub: Grub{
+					Users: []GrubUser{
+						{
+							Name:         "root",
+							PasswordHash: util.StrToPtr("grub.pbkdf2.sha512.10000.874A958E526409..."),
+						},
+					},
+				},
+			},
+			types.Config{
+				Ignition: types.Ignition{
+					Version: "3.4.0-experimental",
+				},
+				Storage: types.Storage{
+					Filesystems: []types.Filesystem{
+						{
+							Device: "/dev/disk/by-label/boot",
+							Format: util.StrToPtr("ext4"),
+							Path:   util.StrToPtr("/boot"),
+						},
+					},
+					Files: []types.File{
+						{
+							Node: types.Node{
+								Path: "/boot/grub2/user.cfg",
+							},
+							FileEmbedded1: types.FileEmbedded1{
+								Contents: types.Resource{
+									Source:      util.StrToPtr("data:,%23%20Generated%20by%20Butane%0A%0Aset%20superusers%3D%22root%22%0Apassword_pbkdf2%20root%20grub.pbkdf2.sha512.10000.874A958E526409...%0A"),
+									Compression: util.StrToPtr(""),
+								},
+							},
+						},
+					},
+				},
+			},
+			[]translate.Translation{
+				{path.New("yaml", "version"), path.New("json", "ignition", "version")},
+				{path.New("yaml", "grub"), path.New("json", "storage")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "filesystems")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "filesystems", 0)},
+				{path.New("yaml", "grub"), path.New("json", "storage", "filesystems", 0, "path")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "filesystems", 0, "device")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "filesystems", 0, "format")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "files")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "files", 0)},
+				{path.New("yaml", "grub"), path.New("json", "storage", "files", 0, "path")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "files", 0, "contents")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "files", 0, "contents", "source")},
+				{path.New("yaml", "grub"), path.New("json", "storage", "files", 0, "contents", "compression")},
+			},
+			report.Report{},
+		},
+		// config with 2 users
+		{
+			Config{
+				Grub: Grub{
+					Users: []GrubUser{
+						{
+							Name:         "root1",
+							PasswordHash: util.StrToPtr("grub.pbkdf2.sha512.10000.874A958E526409..."),
+						},
+						{
+							Name:         "root2",
+							PasswordHash: util.StrToPtr("grub.pbkdf2.sha512.10000.874A958E526409..."),
```

Comment:
It's better to set different hashes, so the tests will notice if we get them mixed up somehow.